modules/prebidServerBidAdapter/index.js

```diff
@@ -930,6 +930,7 @@ const OPEN_RTB_PROTOCOL = {
           if (bid.burl) { bidObject.burl = bid.burl; }
           bidObject.currency = (response.cur) ? response.cur : DEFAULT_S2S_CURRENCY;
           bidObject.meta = bidObject.meta || {};
+          if (bid.ext && bid.ext.dchain) { bidObject.meta.dchain = bid.ext.dchain; }
```

Comment:
Based on the unit test below, dchain is an object, not a stringified object, so I think there's a problem here.
This isn't a deep copy -- believe the right way to copy a full object would be something like this:

```
Object.assign(bidObject.meta.dchain, bid.ext.dchain);
```

@ChrisHuie - please confirm?

Reply:
happy to make that change, but i'm curious why my test would have passed if this is a problem

Reply:
I think I am fully understanding what you are asking. Object.assign doesn't exactly deep copy but it does create a new clone from the source object that copies over the properties and is then allocated differently in memory. The "=" sign, in this case, instead is referencing the source property which acts just like a pointer to the same memory slot. So if bidObject.meta.dchain changes then bid.ext.dchain would also change and vice-versa I believe. With object.assign the changed source property wouldn't affect bid.ext.dchain.

I think object.assign is better for the side effect reason but shouldn't impact these tests

Reply:
i changed to utils.deepClone